### PR TITLE
TMDM-12595 Talend MDM decimal precision error.

### PR DESCRIPTION
**Pull request by yjli-talend**
https://jira.talendforge.org/browse/TMDM-12595
**What is the current behavior?** (You should also link to an open issue here)
Xml schema validate scientific notation value failed.

**What is the new behavior?**
Use plain string value instead of scientific notation.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/handler/ItemTreeHandler.java [hunk 1]
```diff
@@ -211,8 +212,11 @@ private Element buildXML(Document doc, ItemNodeModel currentNodeModel, Map<Strin
                         currentNodeModel.setTypeName(((ForeignKeyBean) value).getConceptName());
                     } else {
                         String valueString = value.toString();
-                        if (value instanceof BigDecimal) {
-                            valueString = ((BigDecimal) value).toPlainString();
+                        if (value instanceof BigDecimal || value instanceof Float || value instanceof Double) {
+                            valueString = new BigDecimal(valueString).toPlainString();
+                            if (value instanceof BigDecimal || value instanceof Float) {
```
Comment (lpw82gf) [inline org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/handler/ItemTreeHandler.java]: Should be `if (value instanceof BigDecimal || value instanceof Float) {`

File: org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/widget/inputfield/propertyeditor/FormatNumberPropertyEditor.java [hunk 2]
```diff
@@ -104,10 +106,19 @@ public String getStringValue(Number value) {
...
-            if (!value.toString().contains(".")) {
-                return value.toString() + ".0";
+        if (type == Float.class || type == Double.class || type == BigDecimal.class) {
+            BigDecimal number = new BigDecimal(value.toString());
+            String valueString = number.toPlainString();
+            if (type == Float.class || type == Double.class) {
+                if (!valueString.contains(".")) { //$NON-NLS-1$
+                    valueString = valueString + ".0"; //$NON-NLS-1$
+                } else {
+                    if (type == Float.class || type == Double.class) {
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/widget/inputfield/propertyeditor/FormatNumberPropertyEditor.java]: The 'if' condition in line 116 is duplicated with line 112, is it needed?

File: org.talend.mdm.core/src/com/amalto/core/storage/record/DataRecordXmlWriter.java [hunk 3]
```diff
@@ -312,8 +312,13 @@ protected void handleSimpleValue(FieldMetadata simpleField, Object value) throws
                     synchronized (TimeConstant.TIME_FORMAT) {
                         out.write((TimeConstant.TIME_FORMAT).format(value));
                     }
-                } else if (Types.DECIMAL.equals(type.getName())) {
-                    out.write(((BigDecimal) value).toPlainString());
+                } else if (Types.FLOAT.equals(type.getName()) || Types.DOUBLE.equals(type.getName())
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core/src/com/amalto/core/storage/record/DataRecordXmlWriter.java]: Could we retrieve this 'else if' into a public method in StorageMetadataUtils.java? I can find a similar code in that class.

File: org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/util/FormatUtil.java [hunk 4]
```diff
@@ -118,4 +118,21 @@ public static Number getDecimalValue(String value, Object franctionDigits) {
         }
         return bigdecimal.setScale(2, RoundingMode.HALF_UP);
     }
+
+    public static String formatFranctionValue(String value) {
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/util/FormatUtil.java]: There is another same name function in StorageMetadataUtils.java, what's the difference?

Comment (yjli-talend) [reply org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/util/FormatUtil.java]: The two method are same,but one is in the gwt client side,the gwt client side can't call the method of back end side.

**Pull request merged**